centreon/www/class/centreonHost.class.php

```diff
@@ -707,25 +705,25 @@ public function replaceMacroInString($hostParam, $string, $antiLoop = null)
          * replace if not template
          */
         if ($row['host_register'] == 1) {
-            if (strpos($string, "\$HOSTADDRESS$")) {
-                $string = str_replace("\$HOSTADDRESS\$", $this->getHostAddress($hostId), $string);
+            if (strpos($string, '$HOSTADDRESS$') !== false) {
+                $string = str_replace('$HOSTADDRESS$', $row['host_address'], $string);
             }
-            if (strpos($string, "\$HOSTNAME$")) {
-                $string = str_replace("\$HOSTNAME\$", $this->getHostName($hostId), $string);
+            if (strpos($string, '$HOSTNAME$') !== false) {
+                $string = str_replace('$HOSTNAME$', $row['host_name'], $string);
             }
-            if (strpos($string, "\$HOSTALIAS$")) {
-                $string = str_replace("\$HOSTALIAS\$", $this->getHostAlias($hostId), $string);
+            if (strpos($string, '$HOSTALIAS$') !== false) {
+                $string = str_replace('$HOSTALIAS$', $row['host_alias'], $string);
             }
-            if (preg_match("\$INSTANCENAME\$", $string)) {
+            if (strpos($string, '$INSTANCENAME$') !== false) {
```

Comment:
⚠️ This new code **is not equivalent** of the previous.
Either this is a fix, and that's ok, you can skip the warning.
Either this wanted to be an equivalent refactor, and **this can be bug prone** !